### PR TITLE
Add paymentMethodSelector to quickcheckout cancelled state

### PR DESCRIPTION
**Pull request by TimFoerster**
When the payment is cancelled by the user, give him the ability to pay again.

image

### CODE REVIEW

File: components/QuickCheckout.php [hunk 1]
```diff
@@ -173,6 +173,8 @@ public function init()
             $this->addComponent(AddressSelector::class, 'shippingAddressSelector', ['type' => 'shipping', 'redirect' => 'quickCheckout']);
         } elseif ($this->step === 'payment') {
```
Comment (mjauvin) [inline components/QuickCheckout.php:174]:
```suggestion
        } elseif ($this->step === 'payment' || $this->step === 'cancelled') {
```

File: components/QuickCheckout.php [hunk 2]
```diff
@@ -173,6 +173,8 @@ public function init()
...
             $this->addComponent(PaymentMethodSelector::class, 'paymentMethodSelector', []);
+        } elseif ($this->step === 'cancelled') {
```
Comment (mjauvin) [inline components/QuickCheckout.php:176]:
```suggestion
```

File: components/QuickCheckout.php [hunk 3]
```diff
@@ -173,6 +173,8 @@ public function init()
...
+            $this->addComponent(PaymentMethodSelector::class, 'paymentMethodSelector', []);
```
Comment (mjauvin) [inline components/QuickCheckout.php:177]:
```suggestion
```

**Review by mjauvin — COMMENTED** [review]
no need to duplicate code here...

File: components/QuickCheckout.php (hunk 3 above)
Comment (mjauvin) [inline components/QuickCheckout.php:177]:
```suggestion
```

**Comment by tobias-kuendig** [comment]
Thank you @TimFoerster and @mjauvin! I really like this idea to allow a second payment attempt after cancelling the payment. Including the component might break the design of existing installations, so I have added the component as a comment/hint in ae8c516. It will be loaded server side so everything is ready to go.

**Pull request closed without merging**